pkg/pillar/cmd/volumemgr/handledownloader.go

```diff
@@ -105,6 +106,24 @@ func MaybeRemoveDownloaderConfig(ctx *volumemgrContext, imageSha string) {
 	log.Functionf("MaybeRemoveDownloaderConfig done for %s", imageSha)
 }
 
+func retryDownload(ctx *volumemgrContext, imageSha string) {
+	m := lookupDownloaderConfig(ctx, imageSha)
+	if m == nil {
+		log.Functionf("retryDownload: config missing for %s",
+			imageSha)
+		return
+	}
+	if m.RefCount == 0 {
+		log.Fatalf("retryDownload: Attempting to retry when "+
+			"RefCount is 0. Image Details - Name: %s, ImageSha: %s, ",
+			m.Name, m.ImageSha256)
+	}
+	m.DoRetry = true
```

Comment:
It is hard to understand how this boolean is used to cause a retry each time the verifier is detecting that the sha256 does not match. Having a RetryCount in both DownloaderConfig and DownloaderStatus would make it be easier to understand (The downloader would cause a download when the count in the config is greater than the count in the status).

Reply:
And it seems like it is required to have a counter which is incremented on each retry function call in order to cause downloader to retry the download more than one time.